src/VecSim/vec_sim_tiered_index.h

```diff
@@ -33,19 +38,26 @@ class VecSimTieredIndex : public VecSimIndexInterface {
     void *jobQueueCtx; // External context to be sent to the submit callback.
     SubmitCB SubmitJobsToQueue;
 
-    void *memoryCtx;
-    UpdateMemoryCB UpdateIndexMemory;
-
     mutable std::shared_mutex flatIndexGuard;
     mutable std::shared_mutex mainIndexGuard;
 
     size_t flatBufferLimit;
 
     void submitSingleJob(AsyncJob *job) {
-        auto **jobs = array_new<AsyncJob *>(1);
-        jobs = array_append(jobs, job);
-        this->SubmitJobsToQueue(this->jobQueue, (AsyncJob **)jobs, 1, this->jobQueueCtx);
-        array_free(jobs);
+        auto destructor = AsyncJobDestructor;
```

Comment:
you can just send it in the function, no?

Reply:
No, as I need to pass an array of function pointers. So I need something to dereference